dsl/src/test/scala/org/http4s/dsl/PathSpec.scala

```diff
@@ -234,6 +235,95 @@ class PathSpec extends Http4sSpec {
       }
     }
 
+    "Matrix extractor" >> {
+      object BoardExtractor
+          extends impl.MatrixVar("square", List("x", "y"), s => Try(s.toInt).toOption)
+
+      object EmptyExtractor extends impl.MatrixVar[List, String]("square", List.empty, s => Some(s))
+
+      "valid" >> {
+        "a matrix var" in {
+          (Path("/board/square;x=42;y=0") match {
+            case Root / "board" / BoardExtractor(x, y) if x == 42 && y == 0 => true
+            case _ => false
+          }) must beTrue
+        }
+
+        "a matrix var mid path" in {
+          (Path("/board/square;x=42;y=0/piece") match {
+            case Root / "board" / BoardExtractor(x, y) / "piece" if x == 42 && y == 0 => true
+            case _ => false
+          }) must beTrue
+        }
+
+        "an empty matrix var but why?" in {
+
+          (Path("/board/square") match {
+            case Root / "board" / EmptyExtractor() => true
+            case _ => false
+          }) must beTrue
+        }
+      }
+
+      "invalid" >> {
+        "empty with semi" in {
+          (Path("/board/square;") match {
+            case Root / "board" / BoardExtractor(x @ _, y @ _) => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "empty without semi" in {
+          (Path("/board/square") match {
+            case Root / "board" / BoardExtractor(x @ _, y @ _) => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "empty with mismatched name" in {
+          (Path("/board/other") match {
+            case Root / "board" / EmptyExtractor() => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "empty axis" in {
+          (Path("/board/square;;y=0") match {
+            case Root / "board" / BoardExtractor(x @ _, y @ _) => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "empty too many = in axis" in {
+          (Path("/board/square;x=42=0;y=9") match {
+            case Root / "board" / BoardExtractor(x @ _, y @ _) => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "unparseable" in {
+          (Path("/board/square;x=42;y=soda") match {
+            case Root / "board" / BoardExtractor(x @ _, y @ _) => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "not enough axes" in {
+          (Path("/board/square;x=42") match {
+            case Root / "board" / BoardExtractor(x @ _, y @ _) => true
+            case _ => false
+          }) must beFalse
+        }
+
+        "to many axes" in {
+          (Path("/board/square;x=42y=0;z=39") match {
```

Comment:
Is this supposed to be `x=42;y=0`?  And why is this invalid?  I might have expected extra axes to be ignored.

Reply:
The similarity to query parameters probably means that they should be. I just need to make the toAssocList stack safe since its currently capped at the number of parameters that are expected to be extracted.